### PR TITLE
Zhasun/grpo add ref logits

### PR DESCRIPTION
**Pull request by mRSun15**
## Summary

Add the ref_log_probs as input. and if the log probs is not none, we don't need to re-calculate it using ref_input.

Add it in corresponding unit test as well.

## Testing Done

- Hardware Type: <BLANK>
- [X] run `make test` to ensure correctness
- [X] run `make checkstyle` to ensure code style
- [X] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by mRSun15** [comment]
@shivam15s  please take a look.
